### PR TITLE
64-bit support for CSGO on Linux and macOS

### PR DESCRIPTION
**Pull request by TheDS**
A lot of the work done for Source2 made this pretty easy. Lots of changes to AMBuild files here though to support building and packaging everything together.

Notable Changes:
- 64-bit binaries live in bin/linux64 or bin/osx64. Doing this allows gameinfo.txt loading to still work if it matters at all.
- Plugins with an .x64 suffix are loaded if no extension is specified
- Added metamod_x64.vdf to package to automatically load 64-bit version of Metamod.

Note: This requires an updated ambuild because it makes use of the ability to change the binary build folder (i.e. BinaryBuilder.localFolder).

### CODE REVIEW

File: core/sourcehook/sh_memory.h [hunk 1]
```diff
@@ -351,6 +351,8 @@ namespace SourceHook
 
 			if (sigaction(SIGBUS, &sa, &osa) == -1)
 				return false;
+			if (sigaction(SIGSEGV, &sa, &osa) == -1)
```
Comment (peace-maker) [inline core/sourcehook/sh_memory.h]: `osa2`!

Comment (TheDS) [reply core/sourcehook/sh_memory.h]: Good catch! Thanks.

File: AMBuildScript [hunk 2]
```diff
@@ -254,11 +266,10 @@ class MMSConfig(object):
         cxx.linkflags += ['-lgcc_eh']
     elif builder.target.platform == 'mac':
       cxx.defines += ['OSX', '_OSX', 'POSIX']
-      cxx.cflags += ['-mmacosx-version-min=10.5']
+      cxx.cflags += ['-mmacosx-version-min=10.9']
       cxx.linkflags += [
-        '-mmacosx-version-min=10.5',
-        '-arch', 'i386',
-        '-lstdc++',
+        '-mmacosx-version-min=10.9',
```
Comment (psychonic) [inline AMBuildScript]: For what it's worth, I believe 10.7 is the min version for CS:GO (as well as TF2 the branch).

Comment (TheDS) [reply AMBuildScript]: I was preparing to release a new srcds binary that bumps the requirement for all games to 10.9. It's necessary to support automatic updates among other things, but I'll put it back to 10.5 for now until that's been released.

**Pull request merged**